pkg/eth/sql.go

```diff
@@ -132,8 +132,8 @@ const (
 							AND receipt_cids.tx_id = transaction_cids.tx_hash
 							AND receipt_cids.header_id = transaction_cids.header_id
 							AND receipt_cids.block_number = transaction_cids.block_number
-							AND transaction_cids.header_id = header_cids.block_hash
```

Comment:
We should also add an `AND transaction_cids.header_id = (SELECT canonical_header_hash(transaction_cids.block_number))` statement to ensure only canonical results (based on internal weighing) are returned.

Reply:
Hmm, I need to examine that quite carefully, as it may have the side-effect of defeating the indexing.

Reply:
Yes, in a quick test against carrion using a (fairly) narrow 500 block range it moved it from an already too slow 4s to whopping 67s.

It's not apples to apples because I had to tweak the query for v4, but the idea is similar.

This could be quite tricky.  Also, is there any particular reason we are open an explicit TX to do this query?  Unless we are dead certain we need one, we should remove it.

Reply:
Curious what the performance imapact is on smaller queries like https://github.com/cerc-io/ipld-eth-server/blob/telackey/249/pkg/eth/sql.go#L98 and https://github.com/cerc-io/ipld-eth-server/blob/telackey/249/pkg/eth/sql.go#L162.